test/kvao/delete-all.suite.js

```diff
@@ -0,0 +1,45 @@
+'use strict';
+
+const bdd = require('../helpers/bdd-if');
+const helpers = require('./_helpers');
+const should = require('should');
+
+module.exports = function(dataSourceFactory, connectorCapabilities) {
+  var supportsDeleteAllOperation =
+    connectorCapabilities.supportsDeleteAllOperation !== false;
+
+  bdd.describeIf(supportsDeleteAllOperation, 'deleteAll', function() {
+    let CacheItem;
+    beforeEach(function unpackContext() {
+      CacheItem = helpers.givenCacheItem(dataSourceFactory);
+    });
+
+    it('removes all keys-value pairs associated to the given model', function() {
+      return helpers.givenKeys(CacheItem, ['key1', 'key2'])
+        .then(() => CacheItem.deleteAll())
+        .then(() => CacheItem.keys())
+        .then((keys) => {
+          should(keys).eql([]);
+        });
+    });
+
+    it('does not remove keys-value pairs unassociated to the given model',
+    function() {
+      var NonCacheItem = dataSourceFactory().createModel('NonCacheItem', {
+        key: String,
+        value: 'any',
```

Comment:
Do we actually need to specify these properties? I think a KV model does not require properties, since the KV methods are accepting arbitrary javascript entity as a value.